File: ruby/bob/bob.rb
class Bob

  def self.hey(remark)

    remark = remark.strip
    return 'Whoa, chill out!' if remark == remark.upcase && remark.count("a-zA-Z") > 0
    return 'Sure.' if remark[-1] =='?' 
    return 'Whatever.' if remark.length > 0
    return 'Fine. Be that way!'
    
  end

end

module BookKeeping
  VERSION = 1
end

File: javascript/bob/bob.js
class Bob {
  hey(sentence) {
    return sentence.toUpperCase() === sentence && sentence.match(/[A-Z]/i) ? sentence.trim().slice(-1) === '?' ? "Calm down, I know what I'm doing!" : 'Whoa, chill out!' : sentence.trim().slice(-1) === '?' ? 'Sure.' : sentence.trim().length === 0 ? 'Fine. Be that way!' : 'Whatever.';
  }
}
module.exports = Bob;